File: 330-patching-array/main.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        
        _sum, cnt, i = 0, 0, 0

        while _sum < n:
            if i < len(nums) and nums[i] <= _sum + 1:
                _sum += nums[i]
                i += 1
            else:
                cnt += 1
                _sum += _sum + 1
        
        return cnt

File: 330-patching-array/main.rs
impl Solution {
    pub fn min_patches(nums: Vec<i32>, n: i32) -> i32 {
        
        let mut sum: i64 = 0;
        let mut cnt = 0;
        let mut i: usize  = 0;

        while sum < n as i64 {
            if i < nums.len() && nums[i] as i64 <= sum + 1 {
                sum += nums[i] as i64;
                i += 1;
            }else {
                cnt += 1;
                sum += sum + 1;
            }
        }

        cnt
    }
}
